pkg/updater/repository/repository.go

```diff
@@ -0,0 +1,324 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package repository contains the packaging logic of the updater.
+package repository
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+)
+
+const (
+	previousVersionLink   = "previous"
+	stableVersionLink     = "stable"
+	experimentVersionLink = "experiment"
+)
+
+// Repository contains the stable and experimental package of a single artifact managed by the updater.
+//
+// On disk the repository is structured as follows:
+// .
+// ├── 7.50.0
+// ├── 7.51.0
+// ├── stable -> 7.50.0 (symlink)
+// └── experiment -> 7.51.0 (symlink)
+//
+// We voluntarily do not load the state of the repository in memory to avoid any bugs where
+// what's on disk and what's in memory are not in sync.
+// All the functions of the repository are "atomic" and ensure no invalid state can be reached
+// even if an error happens during their execution.
+// It is possible to end up with garbage left on disk if an error happens during some operations. This
+// is cleaned up during the next operation.
+type Repository struct {
+	RootPath string
+}
+
+// Status is the status of the repository.
+type Status struct {
```

Comment:
Status holds the stable and and experiment base file. Should we rename `Status` to `Location` or something that signals better what we store in the Struct?

Reply:
Status holds the name/version of stable and experiment to be precise. I did not leak the concept of `version` in repository since it was self-contained in its own package but it makes things a bit more confusing